next-data/providers/blogData.ts

```diff
@@ -0,0 +1,46 @@
+import { cache } from 'react';
+
+import generateBlogData from '@/next-data/generators/blogData.mjs';
+import type { BlogDataRSC } from '@/types';
+
+const blogData = generateBlogData();
+
+const provideBlogData = cache(
+  async (category?: string): Promise<BlogDataRSC> => {
+    return blogData.then(({ posts, categories, pagination }) => {
+      const meta = { categories, pagination };
+
+      if (category && categories.includes(category)) {
+        return {
+          posts: posts.filter(post => post.category === category),
+          pagination: { next: null, prev: null },
+          meta,
+        };
+      }
+
+      if (category && category.startsWith('year-')) {
+        const paramYear = Number(category.replace('year-', ''));
```

Comment:
nit: Wouldn't it be better to use the variable name directly as `year` instead of `paramYear` here 🤔